client/my-sites/pages/virtual-page/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { CompactCard, Gridicon } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ExternalLink } from '@wordpress/components';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch } from 'react-redux';
+import EllipsisMenu from 'calypso/components/ellipsis-menu';
+import InfoPopover from 'calypso/components/info-popover';
+import PopoverMenuItem from 'calypso/components/popover-menu/item';
+import PopoverMenuItemClipboard from 'calypso/components/popover-menu/item-clipboard';
+import { addQueryArgs } from 'calypso/lib/route';
+import { infoNotice } from 'calypso/state/notices/actions';
+import { setLayoutFocus } from 'calypso/state/ui/layout-focus/actions';
+import { setPreviewUrl } from 'calypso/state/ui/preview/actions';
+import { recordEvent } from '../helpers';
+import { PageCardInfoBadge } from '../page-card-info';
+import type { SiteDetails } from '@automattic/data-stores';
+
+interface Props {
+	site: SiteDetails;
+	id: string;
+	type: string;
+	title: string;
+	description?: string;
+	previewUrl?: string;
+	isHomepage?: boolean;
+}
+
+const VirtualPage = ( { site, id, type, title, description, previewUrl, isHomepage }: Props ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const decodedTitle = decodeEntities( title );
+	const editorUrl = addQueryArgs(
+		{ templateId: id, templateType: type },
```

Comment:
Why do we need this? I think without this query param, the homepage still shows the correct template.

Reply:
This might not be required for now. However, I made the component to be `VirtualPage` for general purposes, so we could change to id and type to show another virtual page (template) quickly.

Reply:
I would suggest we delete it. This make the URL ugly (esp. when the user wants to copy / share the homepage edit link)

Reply:
I could delete it if it's the homepage. However, I don't think the URL is ugly. If you go to the site editor, browser all templates, and copy the link from any of the existing templates (screenshot below), you will find that's terrible 😅

![image](https://user-images.githubusercontent.com/13596067/196663932-012ac546-ac22-471f-a5bf-7112fbae94f5.png)

Besides, we have a similar [issue](https://github.com/Automattic/wp-calypso/issues/68955) to show the templates or template parts under the Appearance menu. The link there will also contain something like `postType=wp_template`.

Last, I assume what the edit link looks like is not very important (compared with the view page link), so the current URL is still acceptable.

Reply:
But the template id could be wrong. For example, if the Front Page template is used, your link still takes us to Home template, not Front Page template, which is wrong...

Reply:
> No, the id is correct

Okay, I misunderstood `home_template` prop (as mentioned in the other comment), sorry!

> so the current URL is still acceptable.

I might need to disagree here haha. But not a blocker. @autumnfjeld any comment / opinion here?

Reply:
But users will encounter the same issue when they use a static home page. If they edit the Home page, bookmark a link, and set another page to the homepage, then the bookmarked link will be incorrect either.

Reply:
I made a change to add query args only when it's not homepage to keep it simple 🙂

Reply:
Coming in late here, I assume this is the url your are speaking of is somethign like.  `/wp-admin/site-editor.php?postType=wp_template`    The "ugly" part is `postType=wp_template`?

I guess  I'd say that most users would never even notice the url, and thus functionality is more important.    But I'm only guessing.